spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -247,36 +252,33 @@ public Table createTable(
   public CompletableFuture<WriteApiResponse> writeJsonStream(
       String tableName, InputStream jsonInputStream) {
 
-    CompletableFuture<WriteApiResponse> writeApiFutureResponse =
-        new CompletableFuture<>();
-
-    Thread asyncTask =
-        new Thread(
-            () -> {
-              try {
-                WriteApiResponse apiResponse = getWriteApiResponse(tableName, jsonInputStream);
-                writeApiFutureResponse.complete(apiResponse);
-              } catch (DescriptorValidationException | IOException e) {
-                writeApiFutureResponse.completeExceptionally(e);
-                logger.warn(String.format("Error: %s %n", e.getMessage()), e);
-              } catch (Exception e) {
-                writeApiFutureResponse.completeExceptionally(e);
-                // Restore interrupted state in case of an InterruptedException
-                Thread.currentThread().interrupt();
-              }
-            });
-    asyncTask
-        .start(); // run the thread async so that we can return the writeApiFutureResponse. This
-    // thread can be run in the ExecutorService when it has been wired-in
-
+    CompletableFuture<WriteApiResponse> writeApiFutureResponse = new CompletableFuture<>();
+    Runnable asyncTask =
+        () -> {
+          try {
+            WriteApiResponse apiResponse = getWriteApiResponse(tableName, jsonInputStream);
+            writeApiFutureResponse.complete(apiResponse);
+          } catch (Exception e) {
+            writeApiFutureResponse.completeExceptionally(e);
+            // Restore interrupted state in case of an InterruptedException
+            Thread.currentThread().interrupt();
+            logger.warn("Unable to get write API response.", e);
+          }
+        };
+    ScheduledFuture<?> asyncTaskScheduledFuture =
+        taskScheduler.schedule(asyncTask, Instant.now()); // Run this task
```

Comment:
Can you just call `taskScheduler.execute(asyncTask)`?

Reply:
`TaskScheduler` doesn't expose `execute(...)`. Ref: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/TaskScheduler.html
Down-casting to a concrete implementation might be tricky, as the user might wire a different concrete implementation.

Reply:
Ah true. I missed that.